crates/chia-sdk-driver/src/lib.rs

```diff
@@ -1,20 +1,60 @@
+//! ## Puzzles
+//!
+//! Chia coins have a puzzle, which controls how it can be spent.
+//! The solution is used as the arguments to the puzzle, and the
+//! output is a list of [`Conditions`].
+//!
+//! A puzzle consists of multiple layers composed together.
+//!
+//! ## Layers
+//!
+//! A [`Layer`] is a subset of the logic that makes up a smart coin in Chia.
+//! They are also referred to as "inner puzzles", and the solution can be broken
+//! up into "inner solutions" as well.
+//!
+//! Generally, you can parse and construct the individual layers separately.
+//! This allows them to be composed together freely. However, there are sometimes
+//! additional restraints which limit the ways they can be mixed. For example,
+//! the [`CatLayer`] cannot have another [`CatLayer`] as its inner puzzle, due to the
+//! way it's written. This would create an error when validating the announcements.
+//!
+//! ### P2 Layer
```

Comment:
I think this layer was removed 😅

Reply:
No that's just a description of what a P2 puzzle is. Might be a confusing title